sdks/python/apache_beam/ml/gcp/video_intelligence.py

```diff
@@ -0,0 +1,107 @@
+  ref: https://cloud.google.com/video-intelligence/docs
+  """
+  def __init__(
+      self, features, video_context=None, location_id=None, metadata=None):
+    """
+      Args:
+        features: (List[``videointelligence_v1.enums.Feature``]) Required.
+          the Video Intelligence API features to detect
+        video_context: (dict, ``videointelligence_v1.types.VideoContext``)
+          Optional.
+          Additional video context and/or feature-specific parameters.
+        location_id: (str) Optional.
+          Cloud region where annotation should take place.
+          If no region is specified, a region will be determined
+          based on video file location.
+        metadata: (Sequence[Tuple[str, str]]) Optional.
+          Additional metadata that is provided to the method.
+    """
+    super(AnnotateVideo, self).__init__()
+    self.features = features
+    self.video_context = video_context
+    self.location_id = location_id
+    self.metadata = metadata
+
+  def expand(self, pvalue):
+    return pvalue | ParDo(
+        self._VideoAnnotateFn(
+            features=self.features,
+            video_context=self.video_context,
+            location_id=self.location_id,
+            metadata=self.metadata))
+
+  @typehints.with_input_types(Union[text_type, binary_type])
+  class _VideoAnnotateFn(DoFn):
+    """ A ``DoFn`` that sends every element to the GCP Video Intelligence API
+      and returns a PCollection of
+    ``google.cloud.videointelligence_v1.types.AnnotateVideoResponse``.
+     """
+    def __init__(self, features, video_context, location_id, metadata):
+      super(AnnotateVideo._VideoAnnotateFn, self).__init__()
+      self._client = None
+      self.features = features
+      self.video_context = video_context
+      self.location_id = location_id
+      self.metadata = metadata
+      self.counter = Metrics.counter(self.__class__, "API Calls")
+
+    def start_bundle(self):
+      self._client = helper.get_videointelligence_client()
+
+    def process(self, element, *args, **kwargs):
+      if isinstance(element, text_type):  # Is element an URI to a GCS bucket
+        response = self._client.annotate_video(
+            input_uri=element,
+            features=self.features,
+            video_context=self.video_context,
+            location_id=self.location_id,
+            metadata=self.metadata)
+      elif isinstance(element, binary_type):  # Is element raw bytes
+        response = self._client.annotate_video(
+            input_content=element,
+            features=self.features,
+            video_context=self.video_context,
+            location_id=self.location_id,
+            metadata=self.metadata)
+      else:
+        raise TypeError(
+            "{}: input element needs to be either {} or {}"
+            " got {} instead".format(
+                self.__class__.__name__, text_type, binary_type, type(element)))
+      self.counter.inc()
+      yield response.result(timeout=120)
```

Comment:
Would it make sense to make 120 configurable?

Reply:
Possibly. I haven't used the API enough to say for sure, but we could make it configurable.

Reply:
I think we can make it a default argument to transform with the same default. It would be up to the users to change if they need to.